src/creeps/Zombie.cpp

```diff
@@ -1,61 +1,307 @@
+#include "Node.h"
 #include "Zombie.h"
-#include <math.h>
-#include <random>
-#include "../game/GameManager.h"
 #include "../log/log.h"
-#define PI 3.14159265
-#define ZOMBIE_VELOCITY 200
+#include "../game/GameManager.h"
+#include <utility>
+using namespace std;
 
-Zombie::Zombie(int health, int state) : Movable(ZOMBIE_VELOCITY), health(health), state(state) {
-    setAngle(getRandomAngle());
-    //logv("Create Zombie\n");
+Zombie::Zombie(int health, ZombieState state, int step, int dir, int frame)
+      : Movable(ZOMBIE_VELOCITY), health(health), state(state), step(step),
+        dir(dir), frame(frame) {
+    logv("Create Zombie\n");
 }
 
 Zombie::~Zombie() {
-    //logv("Destroy Zombie\n");
+    logv("Destroy Zombie\n");
+}
+
+/**
+ * Get move direction
+ * Fred Yang
+ * February 14
+ */
+int Zombie::getMoveDir() {
+    if (frame > 0) {
+        return dir;
+    }
+
+    const int sp = getStep();
+    const string pth = getPath();
+    /*
+    cout << "path: " << pth << endl;
+    cout << sp << '-' << pth.length() << endl;
+    */
+
+    return (sp < static_cast<int>(pth.length()) ? stoi(pth.substr(sp,1)) : -1);
 }
 
 void Zombie::onCollision() {
     // Do nothing for now
 }
 
 void Zombie::collidingProjectile(int damage) {
-    health = health - damage;
+    health -= damage;
+}
+
+/*
+ * Returns if the zombie is moving
+ * Robert Arendac
+ * March 7
+*/
+bool Zombie::isMoving() {
+    return (state == ZOMBIE_MOVE);
+}
+
+/*
+ * Does a check to see if the zombie already arrived at the target.
+ * In theory, zombies will only have a movement collision with a target
+ * as their pathfinding should walk around obstacles.
+ * Robert Arendac
+ * March 7
+*/
+bool Zombie::checkTarget() {
+    CollisionHandler &ch = GameManager::instance()->getCollisionHandler();
+    return (ch.detectMovementCollision(this));
```

Comment:
Can be condensed to `return GameManager::instance()->getCollisionHandler().detectMovementCollision(this);`